torch/_subclasses/fake_tensor.py

```diff
@@ -759,10 +759,18 @@ class TensorMetadata:
     sparse_dim: Optional[int]
 
 
-def extract_tensor_metadata(t: torch.Tensor) -> "TensorMetadata":
+def extract_tensor_metadata(t: torch.Tensor) -> TensorMetadata:
     """
     Extract the TensorMetadata of a tensor.
     """
+
+    # WARNING: Any changes to this (probably) need to be reflected in
```

Comment:
I wonder if there's some kind of unit test we could write to alert us if someone adds a field and does not keep them in sync? I guess nothing clever comes to mind.

Actually, all the existing unit tests that validate different hashes for different FakeTensor properties will hit the new codepath, correct? So ideally someone should add a new test for any new field and it would fail if there's not a corresponding change in the cpp?

Reply:
Do we have some kind of "extra debug" flag we can check on the python side? We store the extract_tensor_metadata() in the cache entry - so we could always verify that for a given cache value the metadata  hasn't changed - but I'd only want to do that in some kind of "paranoid" mode since it would defeat a bunch of the perf savings.

Reply:
Oh - I suppose I could attach it to torch._dynamo.config.fake_tensor_cache_crosscheck_enabled which already does a bunch of validation...

Reply:
Oh yeah, that might be a good idea.